src/components/PopupWithForm.js

```diff
@@ -5,6 +5,19 @@ class PopupWithForm extends Popup {
     super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
     this._popupForm = this._popupElement.querySelector(".modal__form");
+    console.log(this._popupElement); // Check if this logs the correct element
+    if (!this._popupElement) {
+      throw new Error(`Element not found for selector: ${selector}`);
+    }
+  }
+
+  setLoading(isLoading) {
+    this._submitBtn = this._popupElement.querySelector(".modal__button");
```

Comment:
Please, find the button in line 11 only once

**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.